src/ast/irbuilderbpf.cpp

```diff
@@ -1478,6 +1477,60 @@ void IRBuilderBPF::CreateAtomicIncCounter(int mapid, uint32_t idx)
   CreateLifetimeEnd(key);
 }
 
+void IRBuilderBPF::CreateAtomicMapInc(Value *ctx,
+                                      Map &map,
+                                      Value *key,
+                                      const location &loc)
+{
+  int mapid = bpftrace_.maps[map.ident].value()->id;
+  CallInst *call = createMapLookup(mapid, key);
+  SizedType &type = map.type;
+
+  Function *parent = GetInsertBlock()->getParent();
+  BasicBlock *lookup_success_block = BasicBlock::Create(module_.getContext(),
+                                                        "lookup_success",
+                                                        parent);
+  BasicBlock *lookup_failure_block = BasicBlock::Create(module_.getContext(),
+                                                        "lookup_failure",
+                                                        parent);
+  BasicBlock *lookup_merge_block = BasicBlock::Create(module_.getContext(),
+                                                      "lookup_merge",
+                                                      parent);
+
+  AllocaInst *value = CreateAllocaBPF(type, "lookup_elem_val");
+  Value *condition = CreateICmpNE(
+      CreateIntCast(call, getInt8PtrTy(), true),
+      ConstantExpr::getCast(Instruction::IntToPtr, getInt64(0), getInt8PtrTy()),
+      "map_lookup_cond");
+  CreateCondBr(condition, lookup_success_block, lookup_failure_block);
+
+  SetInsertPoint(lookup_success_block);
+
+  // createMapLookup  returns an u8*
+  auto *cast = CreatePointerCast(call, value->getType(), "cast");
+
+  // map item is found, increment the value directly
+  CREATE_ATOMIC_RMW(AtomicRMWInst::BinOp::Add,
+                    cast,
+                    getInt64(1),
+                    1,
+                    AtomicOrdering::Monotonic);
```

Comment:
I'm not sure this should be an atomic op. I believe this'll generate code that takes exclusive access to the cache line. Which could be expensive at runtime.

Since `count()` already uses per-cpu maps, we can do a regular add operation on the map value. That should be equivalent and also faster for this use case IIUC.